apollo-router/src/query_planner/bridge_query_planner.rs

```diff
@@ -232,30 +256,53 @@ impl Service<QueryPlannerRequest> for BridgeQueryPlanner {
     }
 
     fn call(&mut self, req: QueryPlannerRequest) -> Self::Future {
+        let QueryPlannerRequest {
+            query,
+            operation_name,
+            context,
+            compiler,
+        } = req;
+
         let this = self.clone();
         let fut = async move {
             let start = Instant::now();
+
+            let file_id = compiler
+                .lock()
+                .await
+                .db
+                .source_file(QUERY_EXECUTABLE.into())
+                .ok_or(QueryPlannerError::SpecError(SpecError::ParsingError(
+                    "missing input file for query".to_string(),
+                )))?;
+            let filtered_query = compiler.lock().await.db.source_code(file_id);
```

Comment:
Let’s lock the mutex once instead of twice:

```suggestion
            let compiler = compiler.lock().await;
            let file_id = compiler
                .db
                .source_file(QUERY_EXECUTABLE.into())
                .ok_or(QueryPlannerError::SpecError(SpecError::ParsingError(
                    "missing input file for query".to_string(),
                )))?;
            let filtered_query = compiler.db.source_code(file_id);
            drop(compiler)
```